Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -45,32 +39,91 @@ extension Command {
                 includeFlag = "-iquote"
             }
             args += [includeFlag, dep.path]
-            args += ["-l\(dep.c99name)"] //FIXME: giving path to other module's -fmodule-map-file is not linking that module
         }
+        return args
+    }
+
+    var linkFlags: [String] {
+        var args: [String] = []
+        for case let dep as ClangModule in dependencies {
+            args += ["-l\(dep.c99name)"]
+        }
+        return args
+    }
 
+    func optimizationFlags(conf: Configuration) -> [String] {
         switch conf {
         case .Debug:
-            args += ["-g", "-O0"]
+            return ["-g", "-O0"]
         case .Release:
-            args += ["-O2"]
+            return ["-O2"]
+        }
+    }
+}
+
+private extension Sources {
+    func compilePathsForBuildDir(wd: String) -> [(filename: String, source: String, object: String, deps: String)] {
+        return relativePaths.map { source in
+            let path = Path.join(root, source)
+            let object = Path.join(wd, "\(source).o")
+            let deps = Path.join(wd, "\(source).d")
+            return (source, path, object, deps)
         }
+    }
+}
+
+extension Command {
+    static func compile(clangModule module: ClangModule, configuration conf: Configuration, prefix: String) -> ([Command], Command) {
+
+        let wd = Path.join(prefix, "\(module.c99name).build")
+        let mkdir = Command.createDirectory(wd)
+
+        ///------------------------------ Compile -----------------------------------------
+        var compileCommands = [Command]()
+        let dependencies = module.dependencies.map{ $0.targetName }
+        let basicArgs = module.basicArgs + module.includeFlags + module.optimizationFlags(conf)
+        for path in module.sources.compilePathsForBuildDir(wd) {
+            var args = basicArgs
+            args += ["-MMD", "-MT", "dependencies", "-MF", path.deps]
+            args += ["-c", path.source, "-o", path.object]
+
+            let node = "<\(module.name).\(path.filename)>"
```

Comment:
Virtual nodes are formed by default when a node name looks like "<NNNN>"... the build system will assume that these do not refer to actual files. See:
  http://llbuild.readthedocs.org/en/latest/buildsystem.html#node-attributes

Using virtual nodes is useful when you want to aggregate a set of actual files to build into a single node you can refer to as an input somewhere else, but in this case we have a real single output so we can just use that.

Your updated hunk does the right thing (and `let node` can go away).